presto-spi/src/main/java/io/prestosql/spi/connector/SchemaTableName.java

```diff
@@ -15,36 +15,67 @@
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import io.prestosql.spi.Name;
 
 import java.util.Objects;
 
+import static io.prestosql.spi.Name.createNonDelimitedName;
+import static io.prestosql.spi.Name.equivalentNames;
 import static io.prestosql.spi.connector.SchemaUtil.checkNotEmpty;
 import static java.util.Locale.ENGLISH;
 
 public class SchemaTableName
 {
-    private final String schemaName;
-    private final String tableName;
+    private final Name schemaName;
+    private final Name tableName;
 
     @JsonCreator
-    public SchemaTableName(@JsonProperty("schema") String schemaName, @JsonProperty("table") String tableName)
+    public SchemaTableName(@JsonProperty("schema") Name schemaName, @JsonProperty("table") Name tableName)
     {
-        this.schemaName = checkNotEmpty(schemaName, "schemaName").toLowerCase(ENGLISH);
-        this.tableName = checkNotEmpty(tableName, "tableName").toLowerCase(ENGLISH);
+        this.schemaName = checkNotEmpty(schemaName, "schemaName");
+        this.tableName = checkNotEmpty(tableName, "tableName");
+    }
+
+    public SchemaTableName(String schemaName, String tableName)
+    {
+        this.schemaName = createNonDelimitedName(checkNotEmpty(schemaName, "schemaName").toLowerCase(ENGLISH));
+        this.tableName = createNonDelimitedName(checkNotEmpty(tableName, "tableName").toLowerCase(ENGLISH));
     }
 
     @JsonProperty("schema")
-    public String getSchemaName()
+    public Name getOriginalSchemaName()
```

Comment:
I forget what we decided to do here, but didn't we discuss using `getSchemaName` for the new "good" name and having a method for the old one (that we'd eventually deprecate)?

Reply:
@martint Yes we were planning to use `getSchemaName` to return the `Name` but we need to first modify all the connectors to use `getLegacySchemaName` and then we can easily rename `getOriginalSchemaName` to the `getSchemaName`